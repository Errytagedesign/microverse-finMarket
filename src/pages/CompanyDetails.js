import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import {
  getCompanyDetails,
  selectCompanies,
} from '../redux/Company/CompanySlice';

export const formatMoney = (figure) => figure.toLocaleString('en-US', {
  style: 'currency',
  // this will make it use $
  currency: 'USD',

  // This will ensure 2 decimal place, even if they are zeros
  minimumFraction: 2,
});

function CompanyDetails() {
  const { id } = useParams();
  const dispatch = useDispatch();

  const { companyDetail } = useSelector(selectCompanies);
  console.log(companyDetail);

  useEffect(() => {
    dispatch(getCompanyDetails(id));
  }, [dispatch, id]);

  return (
    <main className="container mt-5">
      {companyDetail.map(
        ({
          companyName,
          ceo,
          currency,
          exchangeShortName,
          image,
          industry,
          isActivelyTrading,
          mktCap,
          price,
          sector,
          website,
          description,
        }) => (
          <section
            key={ceo}
            className="d-flex flex-column flex-md-row justify-content-between"
          >
            <div className="odd col-12 col-md-4">
              <figure className=" col-12">
                <img src={image} alt={companyName} />
              </figure>
            </div>

            <section className="col-12 col-md-7">
              <div className="even">
                <p> CEO:</p>
                <h3>{ceo}</h3>
              </div>
              <div className="odd d-flex flex-column">
                <p> company description:</p>

                <small>{description}</small>
              </div>
              <div className="even">
                <p> Currency:</p>
                <h3>{currency}</h3>
              </div>
              <div className="odd">
                <p>Exchange Name: </p>
                <h3>{exchangeShortName}</h3>
              </div>
              <div className="even">
                <p>Actively Trading:</p>
                <h3>{isActivelyTrading}</h3>
              </div>
              <div className="odd">
                <p> Price: </p>
                <h3>{`$${price}`}</h3>
              </div>
              <div className="even">
                <p> Market Cap:</p>
                <h3>{formatMoney(mktCap)}</h3>
              </div>
              <div className="odd">
                <p> Industry:</p>
                <h3>{industry}</h3>
              </div>
              <div className="even">
                <p> Sector:</p>
                <h3>{sector}</h3>
              </div>
              <div className="odd">
                <p> Website:</p>
                <button type="button" className="btn btn-primary">
                  {' '}
                  <a href={website} target="_blank" rel="noreferrer">
                    Visit Our Website
                  </a>
                </button>
              </div>
            </section>
          </section>
        ),
      )}
    </main>
  );
}

export default CompanyDetails;
