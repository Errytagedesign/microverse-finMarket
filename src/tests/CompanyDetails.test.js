import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import CompanyDetails from '../pages/CompanyDetails';
import companiesReducer from '../redux/Company/CompanySlice';

describe('CompanyDetails Component', () => {
  let store;

  beforeEach(() => {
    // Create a store with the companiesReducer and the desired initial state
    store = configureStore({
      reducer: {
        companies: companiesReducer,
      },
      preloadedState: {
        companies: {
          companyDetail: [],
        },
      },
    });
  });

  it('renders company details correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <CompanyDetails />
        </BrowserRouter>
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
