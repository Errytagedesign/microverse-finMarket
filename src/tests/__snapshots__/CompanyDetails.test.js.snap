// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyDetails Component renders company details correctly 1`] = `
<div>
  <main
    class=""
  >
    <header
      class="d-flex "
    >
      <section
        class="py-5 d-flex flex-row justify-content-between align-items-center container"
      >
        <a
          href="/"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M11.67 3.87L9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
            />
          </svg>
        </a>
        <div>
           
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.5 6.5A.5.5 0 0 1 4 7v1a4 4 0 0 0 8 0V7a.5.5 0 0 1 1 0v1a5 5 0 0 1-4.5 4.975V15h3a.5.5 0 0 1 0 1h-7a.5.5 0 0 1 0-1h3v-2.025A5 5 0 0 1 3 8V7a.5.5 0 0 1 .5-.5z"
            />
            <path
              d="M10 8a2 2 0 1 1-4 0V3a2 2 0 1 1 4 0v5zM8 0a3 3 0 0 0-3 3v5a3 3 0 0 0 6 0V3a3 3 0 0 0-3-3z"
            />
          </svg>
           
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58a.49.49 0 00.12-.61l-1.92-3.32a.488.488 0 00-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54a.484.484 0 00-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58a.49.49 0 00-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
            />
          </svg>
           
        </div>
      </section>
    </header>
    <div
      class="spinner-border"
    />
  </main>
</div>
`;
